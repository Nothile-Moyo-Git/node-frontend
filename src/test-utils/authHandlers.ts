// handlers.ts
import { graphql } from "msw";

export const handlers = [
  graphql.query("PostUserDetailsResponse", (req, res, ctx) => {
    return res(
      ctx.data({
        PostUserDetailsResponse: {
          sessionCreated: "2024-01-01",
          sessionExpires: "2024-12-31",
          success: true,
          user: {
            _id: "1",
            name: "Nothile Moyo",
            email: "nothile@example.com",
            password: "secret",
            confirmPassword: "secret",
            status: "active",
            posts: [],
          },
        },
      }),
    );
  }),
];
