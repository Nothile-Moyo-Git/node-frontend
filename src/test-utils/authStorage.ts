/**
 * Date Created: 05/08/2025
 *
 * Author: Nothile Moyo
 *
 * Description: This is the local storage handler file for my tests
 * It will handle anything that needs to be set in local storage, in this case, authentication
 *
 * All local storage items are deleted when the request is complete
 */

/**
 * @name setMockAuthStorage
 *
 * @description Mocks setting authentication data for a user
 *
 * @type () => void
 */
export const setMockAuthStorage = ({
  token = "mock-token",
  userId = "mock-user-id",
  expiresIn = new Date(Date.now() + 3600000).toISOString(),
} = {}) => {
  localStorage.setItem("token", token);
  localStorage.setItem("userId", userId);
  localStorage.setItem("expiresIn", expiresIn);
};

export const clearAuthStorage = () => {
  localStorage.removeItem("token");
  localStorage.removeItem("userId");
  localStorage.removeItem("expiresIn");
};
