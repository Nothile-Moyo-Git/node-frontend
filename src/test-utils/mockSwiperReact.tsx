/**
 * Date created : 01/10/2025
 * Author : Nothile Moyo
 *
 * Test routing file, this will create routes which can be used in Jest for testing
 * We use this to be able to render our components
 *
 */

import { ReactElement } from "react";

export const Swiper = (children: ReactElement) => (
  <div data-testid="swiper">{children}</div>
);

export const SwiperSlide = (children: ReactElement) => {
  <div data-testid="swiper-slide">{children}</div>;
};

export default {
  Swiper,
  SwiperSlide,
};
