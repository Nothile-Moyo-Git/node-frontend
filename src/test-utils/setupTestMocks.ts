/**
 * Date created: 30/07/2025
 *
 * Author: Nothile
 *
 * Methods test file, this file defines reusable methods and hooks which can be used across my tests
 * These methods are intended to be used to replicate the official functionality of the app
 *
 */

// Mock useNavigate so it works with jest
export const mockUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockUsedNavigate,
}));
