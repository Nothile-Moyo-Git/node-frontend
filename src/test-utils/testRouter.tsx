/**
 * Date created : 30/07/2025
 * Author : Nothile Moyo
 *
 * Test routing file, this will create routes which can be used in Jest for testing
 * We use this to be able to render our components
 *
 */
import { ReactNode } from "react";
import { render } from "@testing-library/react";
import { createMemoryRouter, RouterProvider } from "react-router-dom";

export function renderWithRouter(ui: ReactNode, { route = "/" } = {}, context) {
  const router = createMemoryRouter(
    [
      {
        path: "/",
        element: ui,
      },
    ],
    { initialEntries: [route] },
  );

  return render(<RouterProvider router={router} />);
}
