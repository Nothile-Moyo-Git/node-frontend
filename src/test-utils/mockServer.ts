import { setupServer } from "msw/node";
import { handlers } from "./authHandlers";

export const server = setupServer(...handlers);
