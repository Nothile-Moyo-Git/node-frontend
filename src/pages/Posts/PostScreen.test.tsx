/**
 * Author: Nothile Moyo
 * Date created: 29/08/2025
 *
 * @description: This is the test file for the Post Screen component
 * We authenticate the user, and if valid, we then show a list of all posts on the page
 */

import "@testing-library/jest-dom";
import {
  clearAuthStorage,
  setMockAuthStorage,
} from "../../test-utils/authStorage";
import { server } from "../../test-utils/mockServer";

// Setup mocks and environment
beforeAll(() => {
  server.listen();
});

// Clear our tests and get mock our fetch so we get the correct ordering
beforeEach(() => {
  setMockAuthStorage();
});

// Cleanup mocks and environment
afterEach(() => {
  clearAuthStorage();
});

//
afterAll(() => {
  server.close();
});
