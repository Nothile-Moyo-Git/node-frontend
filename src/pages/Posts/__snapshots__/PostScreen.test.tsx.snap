// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Screen Component Matches the screenshot 1`] = `
<section
  class="post"
  data-testid="test-id-post-screen"
>
  <h1
    class="post__title"
  >
    2B
  </h1>
  <p
    class="post__date"
  >
    Uploaded: NaN/NaN/NaN NaN:NaN:NaN
  </p>
  <img
    alt="2B"
    class="post__image"
    src="test-file-stub"
  />
  <p>
    YoRHa No. 2 Type B,[a] commonly known as 2B,[b] is a fictional android from the 2017 video game Nier: Automata, a spin-off of the Drakengard series developed by PlatinumGames and published by Square Enix. One of the game's three protagonists, 2B is a soldier for YoRHa, an android task force fighting a proxy war with alien-created Machine Lifeforms. Her story arc focuses on her backstory within YoRHa, and her relationship with her partner 9S, a reconnaissance android. She is also featured in related media, such as the anime Nier: Automata Ver1.1a.
  </p>
</section>
`;
